combat-trainer.lic

```diff
@@ -3530,13 +3542,17 @@ class GameState
   end
 
   def wield_whirlwind_offhand
+    return if twohanded_weapon_skill?
     return unless currently_whirlwinding
     @equipment_manager.wield_weapon_offhand(whirlwind_offhand_name)
   end
 
   def determine_whirlwind_action
-    if @use_weak_attacks && Flags['ct-accuracy-ready']
-      update_analyze_array(true) if @analyze_combo_array.empty?
+    if ( (DRStats.mana < @flame_inner_fire_threshold) && @use_analyze_combos ) || ( @flame_for_expertise && (DRSkill.getxp('Expertise') < 15) )
+      update_analyze_array('flame')
+      @analyze_combo_array.empty? ? 'attack' : @analyze_combo_array.shift
```

Comment:
I just realized these changes are in the `determine_whirlwind_action` method and since I can't whirlwind yet I suppose I thought these changes were for the `melee_attack_verb` method.

Thoughts on adding this logic to `melee_attack_verb` method, too, so barbs who can't/aren't whirlwinding can still take advantage of analyze flame to boost low inner fire?

I've been testing this PR along with the below change (but since I can't whirlwind, I suppose I'm only testing my below change?) and it seems to be working well. I added a new `elsif` condition after checking if whirwinding:
```ruby
  def melee_attack_verb
    if currently_whirlwinding
      determine_whirlwind_action
    # >> START NEW CODE
    elsif DRStats.barbarian? && DRStats.mana < @flame_inner_fire_threshold && @use_analyze_combos
      update_analyze_array('flame')
      @analyze_combo_array.empty? ? 'attack' : @analyze_combo_array.shift
    # << END NEW CODE
    elsif attack_override
      attack_override
    elsif @use_weak_attacks
      brawling? ? 'gouge' : 'jab'
    # !offhand & !brawling are there because you can't use a weapon for brawling commands
    # This issue will cause problems with the combo array system
    elsif @use_analyze_combos && !offhand? && !brawling?
      update_analyze_array if @analyze_combo_array.empty?
      @analyze_combo_array.empty? ? 'attack' : @analyze_combo_array.shift
    else
      'attack'
    end
  end
```

Reply:
I added this but changed it up a bit, and also made sure it would flame to train expertise if that was lacking but IF levels were good enough.

Reply:
Thank you!